scipy/integrate/_cubature.py

```diff
@@ -298,10 +355,45 @@ def cubature(f, a, b, *, rule="gk21", rtol=1e-8, atol=0, max_subdivisions=10000,
 
             rule = ProductNestedFixed([base_rule] * ndim)
 
-    est = rule.estimate(f, a, b, args)
-    err = rule.estimate_error(f, a, b, args)
+    # If any of limits are the wrong way around (a > b), flip them and keep track of
+    # the sign.
+    sign = (-1.0) ** xp.sum(xp.astype(a > b, xp.float64))
+
+    a_flipped = xp.min(xp.stack([a, b]), axis=0)
+    b_flipped = xp.max(xp.stack([a, b]), axis=0)
+
+    a, b = a_flipped, b_flipped
+
+    # If any of the limits are infinite, apply a transformation
+    if xp.any(xp.isinf(a)) or xp.any(xp.isinf(b)):
+        f = _InfiniteLimitsTransform(f, a, b)
+        a, b = f.transformed_limits
+
+        points = [f.inv(point) for point in points]
+        points.extend(f.points)
+
+    # If any problematic points are specified, divide the initial region so that these
+    # points lie on the edge of a subregion.
+    #
+    # This means ``f`` won't be evaluated there if the rule being used has no evaluation
+    # points on the boundary.
+    if points == []:
```

Comment:
prehaps more idiomatic?
```suggestion
    if len(points) == 0:
```